### PR TITLE
chore: improve `HashedStorage`

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

**Review by rkrasiuk — COMMENTED** [review]
thanks, this is cleaner

File: crates/trie/src/hashed_cursor/post_state.rs [hunk 1]
```diff
@@ -54,16 +54,28 @@ impl HashedStorage {
...
     }
+
+    /// Insert storage entry.
+    #[inline]
+    pub fn insert_storage(&mut self, slot: B256, value: U256) {
+        if value.is_zero() {
+            self.insert_zero_valued_slot(slot)
+        } else {
+            self.insert_non_zero_valued_storage(slot, value)
+        }
```
[the remark this answers is not in the document]
Comment (yjhmelody) [reply crates/trie/src/hashed_cursor/post_state.rs:77]: Seems some tests still need these functions.

[the remark this answers is not in the document]
Comment (yjhmelody) [reply crates/trie/src/hashed_cursor/post_state.rs]: Ok, I make these funcs private.

**Review by onbjerg — APPROVED** [review]
lgtm

File: crates/trie/src/hashed_cursor/post_state.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rkrasiuk) [reply crates/trie/src/hashed_cursor/post_state.rs]: @yjhmelody let's remove them altogether

**Comment by rkrasiuk** [comment]
let's get rid of the extra functions as suggested above. PR will need to be rebased once https://github.com/paradigmxyz/reth/pull/6066 is merged

File: crates/trie/src/state.rs [hunk 2]
```diff
@@ -258,15 +254,15 @@ impl HashedStorage {
...
-    pub fn insert_zero_valued_slot(&mut self, slot: B256) {
-        self.zero_valued_slots.insert(slot);
+    /// Insert storage entry.
+    #[inline]
+    pub fn insert_storage(&mut self, slot: B256, value: U256) {
+        if value.is_zero() {
+            self.zero_valued_slots.insert(slot);
+        } else {
+            debug_assert!(value != U256::ZERO, "value cannot be zero");
+            self.non_zero_valued_storage.push((slot, value));
```
Comment (rkrasiuk) [inline crates/trie/src/state.rs:264]: this is a noop
```suggestion
        } else {
            self.non_zero_valued_storage.push((slot, value));
```

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**